onnx/src/commonMain/kotlin/org/jetbrains/kotlinx/dl/api/inference/onnx/facealignment/FaceDetectionModelBase.kt

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright 2022 JetBrains s.r.o. and Kotlin Deep Learning project contributors. All Rights Reserved.
+ * Use of this source code is governed by the Apache 2.0 license that can be found in the LICENSE.txt file.
+ */
+
+package org.jetbrains.kotlinx.dl.api.inference.onnx.facealignment
+
+import org.jetbrains.kotlinx.dl.api.inference.objectdetection.DetectedObject
+import org.jetbrains.kotlinx.dl.api.inference.onnx.OnnxHighLevelModel
+import java.lang.Float.min
+import kotlin.math.max
+
+/**
+ * Base class for face detection models.
+ */
+public abstract class FaceDetectionModelBase<I> : OnnxHighLevelModel<I, List<DetectedObject>> {
+
+    override fun convert(output: Map<String, Any>): List<DetectedObject> {
+        val scores = (output["scores"] as Array<*>)[0] as Array<FloatArray>
+        val boxes = (output["boxes"] as Array<*>)[0] as Array<FloatArray>
+
+        if (scores.isEmpty()) return emptyList()
+
+        val result = mutableListOf<DetectedObject>()
+        for (classIndex in 1 until scores[0].size) {
+            for ((box, classScores) in boxes.zip(scores)) {
+                val score = classScores[classIndex]
+                if (score > THRESHOLD) {
+                    result.add(DetectedObject(box[0], box[2], box[1], box[3], score))
+                }
+            }
+        }
+        return result
+    }
+
+    /**
+     * Detects [topK] faces on the given [image]. If [topK] is negative all detected faces are returned.
+     * @param [iouThreshold] threshold IoU value for the non-maximum suppression applied during postprocessing
+     */
+    public fun detectFaces(image: I, topK: Int = 5, iouThreshold: Float = 0.5f): List<DetectedObject> {
+        val detectedObjects = predict(image)
+        return suppressNonMaxBoxes(detectedObjects, topK, iouThreshold)
+    }
+
+    private companion object {
+        private const val THRESHOLD = 0.7
+        private const val EPS = Float.MIN_VALUE
+
+        private fun suppressNonMaxBoxes(boxes: List<DetectedObject>,
```

Comment:
Making NMS a public utility function makes sense because it's common post-processing for detector models where NMS is not part of the model graph, and users can use it with their custom models.